### PR TITLE
JPA Entity Filtering option does not work when using reactive option

### LINKED ISSUE #14616: JPA Entity Filtering option does not work when using reactive option
##### **Overview of the issue**

If you select reactive = true and also try to enable filtering on any entities, those entities will not have any filtering code generated for them, or at least the API does not support it

##### **Motivation for or Use Case**

If you want to have entity filtering and reactive at same time

##### **Reproduce the error**

Enable filtering on any entity and turn reactive on

##### **Related issues**

None seen

##### **Suggest a Fix**

Unsure

##### **JHipster Version(s)**

7.0.1

##### **JHipster configuration**

JDL to quickly reproduce, Just toggle reactive on and off and check resulting generation

<details>
  <summary>JDL definitions</summary>
  <pre>
application {
  config {
    applicationType monolith
    authenticationType jwt
    baseName reactivefilterbug
    buildTool gradle
    packageName com.foo.bar
    reactive true
  }

  entities *
}

entity A {
  a Boolean required
  b Integer required
  c String required
}

filter *
  </pre>
  </details>

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

Generated from JDL above

##### **Browsers and Operating System**

Should not be browser or platform related but testing on Fedora 33

### PR DESCRIPTION
**Pull request by dwarakaprasad**
This PR fixes #14616 
---

Please make sure the below checklist is followed for Pull Requests.

- [X] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [X] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [X] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

fix #14616 
Also related to jhipster/jhipster-bom#1003

### CODE REVIEW

File: test/support/entity-samples.mts [hunk 1]
```diff
@@ -5,6 +5,7 @@ const { UUID } = CommonDBTypes;
 export const entitySimple = {
   name: 'Simple',
   changelogDate: '20220129000100',
+  jpaMetamodelFiltering: true,
```
Comment (dwarakaprasad) [inline test/support/entity-samples.mts:8]: Added this here to have both positive testing for sql option and negative testing for non-sql option

**Comment by Tcharl** [comment]
Looks awesome! Could you please regenerate the snapshot so that we'll merge?

**Comment by mshima** [comment]
https://github.com/jhipster/jhipster-bom/pull/1003 Needs to be merged first and the bom repo needs to be updated before merging this

**Comment by dwarakaprasad** [comment]
> @dwarakaprasad Can you fix the conflicts?

Yup, will be taking care of this today...

**Comment by dwarakaprasad** [comment]
> This branch has no conflicts with the base branch

@DanielFran done

**Review by DanielFran — APPROVED** [review]

**Comment by dwarakaprasad** [comment]
bug bounty claim [https://opencollective.com/generator-jhipster/expenses/160398](https://opencollective.com/generator-jhipster/expenses/160398)

**Pull request merged**